crates/noirc_frontend/src/graph/mod.rs

```diff
@@ -83,7 +83,13 @@ impl CrateGraph {
     pub fn add_crate_root(&mut self, crate_type: CrateType, file_id: FileId) -> CrateId {
         let mut roots_with_file_id =
             self.arena.iter().filter(|(_, crate_data)| crate_data.root_file_id == file_id);
-        assert!(roots_with_file_id.next().is_none(), "you cannot add the same file id twice");
+
+        let next_file_id = roots_with_file_id.next();
+        assert!(next_file_id.is_none(), "you cannot add the same file id twice");
+
+        if let Some(file_id) = next_file_id {
+            return *file_id.0;
+        }
```

Comment:
Looks like this doesn't do anything. We assert the Option is None first and if not, we panic. Then we check if it is Some afterward, although we already would have panicked by this point.

Reply:
i've removed this check